desc/objectives/_fast_ion.py

```diff
@@ -359,3 +360,263 @@ def _compute_spline(self, params, constants=None):
+            self._grid = LinearGrid(M=eq.M_grid, N=eq.N_grid, NFP=eq.NFP, sym=False)
+        assert self._grid.can_fft2
+        self._constants["clebsch"] = FourierChebyshevSeries.nodes(
+            self._X,
+            self._Y,
+            self._grid.compress(self._grid.nodes[:, 0]),
+            domain=(0, 2 * np.pi),
+        )
+        self._constants["fieldline quad"] = leggauss(self._hyperparam["Y_B"] // 2)
+        self._constants["quad"] = get_quadrature(
+            leggauss(self._hyperparam.pop("num_quad")),
+            (automorphism_sin, grad_automorphism_sin),
+        )
+        self._dim_f = self._grid.num_rho
+        self._target, self._bounds = _parse_callable_target_bounds(
+            self._target, self._bounds, self._grid.compress(self._grid.nodes[:, 0])
+        )
+        self._constants["transforms"] = get_transforms(self._key, eq, grid=self._grid)
+        self._constants["profiles"] = get_profiles(self._key, eq, grid=self._grid)
+        super().build(use_jit=use_jit, verbose=verbose)
+
+    def compute(self, params, constants=None):
+        """Compute dJ_ds.
+
+        Parameters
+        ----------
+        params : dict
+            Dictionary of equilibrium degrees of freedom, e.g.
+            ``Equilibrium.params_dict``.
+        constants : dict
+            Dictionary of constant data, e.g. transforms, profiles etc.
+            Defaults to ``self.constants``.
+
+        Returns
+        -------
+        dJ_ds : ndarray
+            maxJ proxy a function of the flux surface label.
+
+        """
+        if constants is None:
+            constants = self.constants
+        eq = self.things[0]
+        data = compute_fun(
+            eq, "iota", params, constants["transforms"], constants["profiles"]
+        )
+        # TODO (#1034): Use old theta values as initial guess.
+        theta = Bounce2D.compute_theta(
+            eq,
+            self._X,
+            self._Y,
+            iota=constants["transforms"]["grid"].compress(data["iota"]),
+            clebsch=constants["clebsch"],
+            # Pass in params so that root finding is done with the new
+            # perturbed λ coefficients and not the original equilibrium's.
+            params=params,
+        )
+        data = compute_fun(
+            eq,
+            self._key,
+            params,
+            constants["transforms"],
+            constants["profiles"],
+            data,
+            theta=theta,
+            alpha=constants["alpha"],
+            fieldline_quad=constants["fieldline quad"],
+            quad=constants["quad"],
+            **self._hyperparam,
+        )
+        dJ_ds = constants["transforms"]["grid"].compress(data[self._key])
+
+        thresh0, w0, w1 = constants["thresh0"], constants["w0"], constants["w1"]
+
+        # Shifted ReLU operation
+        dJ_ds_filtrd = (dJ_ds - thresh0) * (dJ_ds >= thresh0)
+        results = w0 * jnp.sum(dJ_ds_filtrd) + w1 * jnp.max(dJ_ds_filtrd)
```

Comment:
The docstring suggests that the sum and max should be performed along axes (-1 and -2) but not axis 0.

Reply:
Yes, no sum over rho. Only alpha and pitch.